patroni/postgresql.py

```diff
@@ -698,31 +690,28 @@ def start(self, block_callbacks=False):
 
         self.set_state('starting')
 
-        env = {'PATH': os.environ.get('PATH')}
-        # pg_ctl will write a FATAL if the username is incorrect. exporting PGUSER if necessary
-        if 'username' in self._superuser and self._superuser['username'] != os.environ.get('USER'):
-            env['PGUSER'] = self._superuser['username']
-
         self._write_postgresql_conf()
         self.resolve_connection_addresses()
 
-        options = ' '.join("--{0}='{1}'".format(p, self._server_parameters[p]) for p, v in self.CMDLINE_OPTIONS.items()
+        options = ' '.join('--{0}="{1}"'.format(p, self._server_parameters[p]) for p, v in self.CMDLINE_OPTIONS.items()
```

Comment:
https://github.com/zalando/patroni/blob/master/patroni/postgresql.py#L58

Only parameters defined in `CMDLINE_OPTIONS` will be passed via command line.
Also it contains function for checking values of parameters. So it should be safe.